vulkaninfo/CMakeLists.txt

```diff
@@ -11,7 +11,35 @@ else()
     add_executable(vulkaninfo vulkaninfo.c)
 endif()
 
-if(APPLE)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    include(FindPkgConfig)
+    option(BUILD_WSI_XCB_SUPPORT "Build XCB WSI support" ON)
+    option(BUILD_WSI_XLIB_SUPPORT "Build Xlib WSI support" ON)
+    set(VULKANINFO_WSI_SELECTION "XCB" CACHE STRING "Select WSI target for vulkaninfo (XCB, XLIB)")
+
+    if (BUILD_WSI_XCB_SUPPORT)
+        find_package(XCB REQUIRED)
+    endif()
+
+    if (BUILD_WSI_XLIB_SUPPORT)
+        find_package(X11 REQUIRED)
+    endif()
+    
+    if (VULKANINFO_WSI_SELECTION STREQUAL "XCB")
+        if (NOT BUILD_WSI_XCB_SUPPORT)
+            message( FATAL_ERROR "Selected XCB for vulkaninfo build but not building Xcb support" )
+        endif()
+        target_include_directories(vulkaninfo PRIVATE ${XCB_INCLUDE_DIRS})
+        target_link_libraries(vulkaninfo ${XCB_LIBRARIES})
+        target_compile_definitions(vulkaninfo PRIVATE -DVK_USE_PLATFORM_XCB_KHR)
+    elseif (VULKANINFO_WSI_SELECTION STREQUAL "XLIB")
+        if (NOT BUILD_WSI_XLIB_SUPPORT)
+            message( FATAL_ERROR "Selected XLIB for vulkaninfo build but not building Xlib support" )
+        endif()
+        target_include_directories(vulkaninfo PRIVATE ${X11_INCLUDE_DIRS})
```

Comment:
Still not right.   See the FindX11.cmake documentation and/or the code for this Module in your CMake installation.

You can/should check your work by _temporarily_ inserting a `MESSAGE(STATUS "path is ${X11_INCLUDE_DIRS})` command to see if there is anything there.

If you tested this and it happened to work, it may be because the X11 include files happen to be in a default system search path (e.g., /usr/include).  This won't always be the case.